app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala

```diff
@@ -39,32 +39,43 @@ class NationalInsuranceRecordService @Inject()(
   featureFlagService: FeatureFlagService
 ) {
 
-  def getNationalInsuranceRecord(nino: Nino)(implicit hc: HeaderCarrier): Future[Either[ExclusionResponse, NationalInsuranceRecord]] =
+  def getNationalInsuranceRecord(nino: Nino)(implicit hc: HeaderCarrier): Future[Either[DesError, NationalInsuranceRecordResult]] =
     featureFlagService.get(ProxyCacheToggle) flatMap {
       proxyCache =>
         citizenDetailsService.checkManualCorrespondenceIndicator(nino) flatMap {
           mci =>
             if (proxyCache.isEnabled) {
-              proxyCacheConnector.get(nino) map {
-                pcd =>
-                  buildNationalInsuranceRecord(pcd.niRecord, pcd.summary, pcd.liabilities, mci)
-              }
+              proxyCacheConnector.get(nino) map (_.map (pcd => buildNationalInsuranceRecord(pcd.niRecord, pcd.summary, pcd.liabilities, mci)))
```

Comment:
Similar question here, but guessing this returns an Either and is right biased?

Reply:
Yes